### PR TITLE
Feature/unverified user status

### PR DESCRIPTION
**Pull request by JoshuaHinman**
# Description

Created new decorator `@Unverified` to allow permission for certain routes to unverified users
Also returns 403 forbidden for unverified users when accessing other routes
Added e2e tests 

## Issue link

Fixes #[ (86b0rfgv4)](https://app.clickup.com/t/86b0rfgv4)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested in Swagger:
Log in with: yoshi@gmail.com , password
Routes:
- /auth/resend-email
- /users/me
- /auth/logout
- /auth/verify - email can't be manually tested without verification token, but is covered in e2e test
- /voyages/features/feature-categories - for 403 response

e2e: `yarn test:e2e unverifiedUser`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: test/unverifiedUser.e2e-spec.ts
Comment (cherylli) [inline test/unverifiedUser.e2e-spec.ts:1]: good idea for making a new e2e file for this

**Review by cherylli — COMMENTED** [review]
Everything looks good, I ran all the tests, reseeded database, manually checked on swagger.

I think we should also update the readme under ##Custom Decorators

**Review by cherylli — APPROVED** [review]
thanks!

**Review by timDeHof — APPROVED** [review]
everything looks great! tested on swagger with the suggested test routes and ran `yarn test:e2e unverifiedUser`. Both passed!

**Pull request merged**